src/store/flow/flow.ts

```diff
@@ -579,13 +581,17 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
     state.selectedBlocks = []
   },
 
-  async flow_duplicateAllSelectedBlocks({state, dispatch}) {
-    const newBlocksUuid: string[] = []
-    forEach(state.selectedBlocks, async (blockId: IBlock['uuid']) => {
-      const duplicatedBlock: IBlock = await dispatch('flow_duplicateBlock', {blockId})
-      newBlocksUuid.push(duplicatedBlock.uuid)
-    })
-    state.selectedBlocks = newBlocksUuid
```

Comment:
This code was not awaiting for individual `flow_duplicateBlock` actions.
So returning `newBlocksUuid` from this function was giving an empty (at least initially) array.